docs/docs/using-pants/key-concepts/targets-and-build-files.mdx

```diff
@@ -60,6 +60,31 @@ python_distribution(
 )
 ```
 
+### Multiple BUILD files in a directory
+
+Typically, there would be one BUILD file in every directory containing source code and any other resources you
+may want to use as part of your builds. However, you can have multiple BUILD files in a single directory.
+When running a Pants goal, the contents of the BUILD files will be merged making it possible to better group your
+targets, if desired. For instance, you could store source code targets and
+[macros](../../writing-plugins/macros.mdx) in different files to have better separation of concerns:
```

Comment:
Link tested.

Reply:
> For instance, you could store source code targets and
[macros](../../writing-plugins/macros.mdx) in different files to have better separation of concerns

The example seems to show having different targets in the files, which works fine, but the above text talks about macros, and that will not work. The files are not merged before parsing, so if you decalare a `def my_macro(): ...` in one, you can't call it from the other.

Reply:
That is, they're parsed, and processed, individually, they will just share "namespace" for their respective targets, so they show up together..

Reply:
Thanks, Andreas! Perhaps bringing macros into this wasn't worth it. I've updated the docs to use files and code targets; this should keep things simple.